### PR TITLE
feat(liga): implement GET  ligas ranking (Task #307)

### LINKED ISSUE #307: Ranking list of participants from highest to lowest points
Replaces the stub with the real ranking logic: retrieves all Liga entries with their associated user data, orders them by points descending, and assigns sequential positions. Feature tests are included.

#### Context:
As a user,
I want to see the list of all participants ordered from highest to lowest points,
so that I can know my position and compare myself with other users in the ranking list.

#### Additional Context:
This task replaces the hardcoded stub in ranking()

#### Acceptance Criteria:
- GET /api/ligas/ranking returns entries sorted by points descending
- Each entry contains: position (1-based sequential), user_name,points, timestamps
- Position numbers are correct: highest points = position 1
- Returns an empty array [] with 200 when no entries exist
- No authentication token is required to call this endpoint
- Feature test passes: php artisan test --filter GetRankingTest

#### To do:
- Replace the stub ranking() method in LigaController  that eager-loads the related user, orders by points descending, and maps the result to inject a sequential position field starting at 1
- Create a feature test under backend/src/tests/Feature/Liga/ covering: empty table returns [], correct sort order, correct position sequence, ... (to be defined)

#### Risks:
- The stub in Task #299 returns a hardcoded user_id in the response. Confirm whether user_id should also be included in the real response — the frontend +5 button (ticket #363) needs it to call PUT /api/ligas/{user}/points

### PR DESCRIPTION
**Pull request by fdesouzabcn**
### Summary
Implements the (real) public ranking endpoint for the Liga feature.
- GET /api/ligas/ranking — returns all liga entries ordered by points descending, with sequential positions computed at query time
- Endpoint is public
- Response shape: [{ position, user_id, points, created_at, updated_at }]
- 7 tests Passed — covering happy path and edge cases

### Important:
This PR cherry-picks pending dependencies from branches awaiting merge, to unblock this implementation:

Commit | Source | Task
-- | -- | --
88ca4af | 302-model-creation | Liga model + User ligaEntry()
dce5ab6 | feature/liga-factory-seeder | LigaFactory + LigaSeeder
906777b , 9909c20 | feature/liga-endpoints-stub | LigaController stub + routes

#### How to test locally
```
git checkout feature/307-ranking-list-of-participants
docker compose up -d
docker compose exec backend php artisan migrate
docker compose exec backend php artisan test --filter LigaGetRankingTest
```
#### Merge conflict notes for reviewer
I'm aware that when 302-model-creation, feature/liga-factory-seeder, and feature/liga-endpoints-stub merge into develop, this branch will have conflicts on cherry-picked files (Liga.php, User.php, LigaFactory.php, api.php, LigaController.php). But it should be easily resolvable by resolving the merge conflicts. The only that i expect a major change is the feature/liga-factory-seeder that requires few updates. 

### Commits Note 

- commit 341d1bc (wip: save progress...) is a save-point commit with no functional change — safely ignore during code review.

### CODE REVIEW

**Review by laisrod — COMMENTED** [review]
Hola,

El endpoint GET /api/ligas/ranking estan bien. Los tests cubren bien los casos principales. Dejo algunos puntos a revisar:

1. Rutas de escritura sin autenticación

PUT /ligas/{user}/points y POST /ligas son rutas públicas. Cualquiera podría añadir puntos sin estar autenticado. Deberían estar protegidas con auth:sanctum.

2. addPoints y store son stubs sin implementación real

public function addPoints(User $user): JsonResponse
{
    return response()->json(['user_id' => $user->id, 'points' => 99]); // hardcoded
}
Está bien para este sprint, pero debería quedar documentado como pendiente para no olvidarlo.

3. LigaSeeder usa rand() en lugar de fake()

Inconsistente con LigaFactory que ya usa fake()->numberBetween(). Mejor unificar.

4. Ficheros sin newline al final

LigaFactory.php, LigaSeeder.php y LigaStubTest.php no tienen newline al final — es una buena práctica en PHP.

5. Test duplicado

LigaStubTest y LigaGetRankingTest tienen el mismo test de accesibilidad del endpoint. Se puede eliminar uno.

En general muy buen trabajo, el endpoint de ranking está listo para integrarlo. Gracias!

**Comment by fdesouzabcn** [comment]
@laisrod, gracias por tu revisión. Solicito nuevamente la aprobación, ya que los comentarios parecen más relacionados con PRs de otras tareas. A continuación, mis comentarios sobre tus puntos.

> 1. Rutas de escritura sin autenticación

Para este sprint, fue acordado con que todas las rutas serían públicas.

### Creo que los siguientes comentarios corresponden mejor a la tarea #382, por lo que etiqueto a @HebeStark para su conocimiento.

> 1. Rutas de escritura sin autenticación
PUT /ligas/{user}/points y POST /ligas son rutas públicas. Cualquiera podría añadir puntos sin estar autenticado. Deberían estar protegidas con auth:sanctum.

Es muy buen punto, pero como dicho anteriormente, para este srpint hemos acordado hacer todas las rutas publicas. Ademas 
Creo que esto comentario corresponderia mas al PR de la tarea #382, asi que hago un tag a @HebeStark aqui. 

> 2. addPoints y store son stubs sin implementación real

Esto deberia ser actualizado con la PR de la tarea #382


### Dado que utilicé cherry-pick de la PR #381 para avanzar con esta PR, los siguientes comentarios corresponden más a la tarea #304 (PR #381). Se etiqueta a @AlejandroJarque para su conocimiento.

>3. LigaSeeder usa rand() en lugar de fake()
Inconsistente con LigaFactory que ya usa fake()->numberBetween(). Mejor unificar.

Muy buen punto y este comentario también lo dio otro colega en #381 

> 4. Ficheros sin newline al final
LigaFactory.php, LigaSeeder.php y LigaStubTest.php no tienen newline al final — es una buena práctica en PHP.

> Test duplicado
LigaStubTest y LigaGetRankingTest tienen el mismo test de accesibilidad del endpoint. Se puede eliminar uno.

¡Excelente observación! Tomamos nota y asegurémonos de modificarlos con una próxima PR que hagamos mas adelante. 


Una vez más, gracias por tu revision.

**Review by AlejandroJarque — APPROVED** [review]

**Review by laisrod — COMMENTED** [review]
Hola Flavio,

Gracias por las aclaraciones. Entiendo que los puntos sobre autenticación y los stubs son decisiones de sprint y corresponden a otras PRs.

Dicho esto, hay algunos puntos menores que sí pertenecen a esta PR y que se pueden corregir antes del merge:

1. **LigaSeeder.php** — usa `rand()` en lugar de `fake()->numberBetween()`. La factory ya usa `fake()`, mejor unificar aquí también.

2. **Newline al final de ficheros** — `LigaFactory.php`, `LigaSeeder.php` y `LigaStubTest.php` no tienen newline al final. Es una convención estándar en PHP.

3. **Test duplicado** — `LigaStubTest::test_ranking_endpoint_is_accessible()` comprueba exactamente lo mismo que el primer test de `LigaGetRankingTest`. Se puede eliminar de `LigaStubTest`.

4. **User.php** — hay 3 líneas en blanco seguidas después de los imports, y la indentación de `ligaEntry()` usa 2 espacios en lugar de 4 (inconsistente con el resto de la clase).

5. **Liga.php** — hay una línea en blanco de más antes del cierre `}`.

Son cambios pequeños pero mantienen la consistencia del código. Una vez aplicados, apruebo sin problema.

Gracias!

**Review by albertvallsbe — APPROVED** [review]

**Review by kenan-it-academy — CHANGES_REQUESTED** [review]
This PR is WAY too big. Does it accidentally contain code from other branches?

**Comment by fdesouzabcn** [comment]
> This PR is WAY too big. Does it accidentally contain code from other branches?

@kenan-it-academy ,As explained in the PR request (comment #1), I cherry-picked to recreate the necessary files, resulting in commits from other branches. My changes only affect:

- backend/src/app/Http/Controllers/LigaController.php (lines 9 and 14-28)
- backend/src/tests/Feature/LigaTest/LigaGetRankingTest.php (entire file)

The other files were expected to be updated later via merge conflicts.  
> And replying to you I realized I should have explained this even more clearly in my first comment.

Having explained that, and haven listed your POV regarding cherry-picking usage in this project, my decision rests on these points:

1. If this PR is not approved, I will wait for the "dependent" PRs (#382 and #381) to be approved and merged, then modify this PR once.

1.If this explanation is acceptable and you're willing to make an exception to the cherry-picking rule, my colleagues and I can resolve the merge conflicts later.

I'm also open to your suggestions.

**Comment by fdesouzabcn** [comment]
@kenan-it-academy  todos conflictos ya solucionados. Faltando solo tu aprobacion

**Review by HebeStark — APPROVED** [review]
Aprovado despues de resolver conflictos en equipo por video llamada

**Review by HebeStark — APPROVED** [review]
Aprobado después de resolver conflictos en equipo por video llamada

File: backend/src/tests/Feature/LigaTest/LigaGetRankingTest.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+use App\Models\Liga;
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+class LigaGetRankingTest extends TestCase
+{
+    use RefreshDatabase;
+
+
+    public function test_endpoint_is_public_and_requires_no_auth_token(): void
```
Comment (kenan-it-academy) [inline backend/src/tests/Feature/LigaTest/LigaGetRankingTest.php:17]: I think this test is redundant and essentially absorbed by the other tests, to be honest

**Review by kenan-it-academy — APPROVED** [review]
I think there's a tiny redundant test, but I don't consider that blocking as it's not a duplicate. Code looks great!
